internal/pkg/encoding/jsonnet/vm.go

```diff
@@ -0,0 +1,69 @@
+package jsonnet
+
+import (
+	"bytes"
+	"encoding/json"
+	"sync"
+
+	"github.com/google/go-jsonnet"
+	"github.com/google/go-jsonnet/ast"
+	"github.com/google/go-jsonnet/parser"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+)
+
+type VM struct {
+	lock    sync.Mutex
+	vm      *jsonnet.VM
+	err     error
+	payload any
+}
```

Comment:
Is the `encoding/jsonnet` right layer for the pool implementation?

--------------

My point of view:
- forked lib: `Freeze` method :heavy_check_mark: :+1: 
- `encoding/jsonnet`: (almost) no changes are necessary
- `receive/jsonnet`: implementation of the `pool`, wrapped `VM`
   - `payload` may be a specific type, for example `receivectx.Context`
   - if we would need another performant implementation of the Jsonnet, we can repeat this way in another pkg, with another payload type.
   
-----------------

I really like how you designed it, I think there is only one step left ^^^. 

This way we can get rid of retyping `payload` in every function <- that is always a last choice, if another approach cannot be used.

Reply:
You can also consider a generic implementation, where the payload will be taken from the generic type `[P any]`.

Reply:
> pool moved to receive/jsonnet?
no, generic pool can be in `encoding/jsonnet`